tests/integration/galley/webhook_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package galley
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+)
+
+var (
+	vwcName    = "istio-galley"
+	deployName = "istio-galley"
+	istioNs    = "istio-system"
+	sleepDelay = 3 * time.Second // How long to wait to give the reconcile loop an opportunity to act
+)
+
+var removeGalleyYaml = fmt.Sprintf(`
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: %s
+  namespace: %s
+`, deployName, istioNs)
+
+// Verify that scaling up/down doesn't modify webhook configuration
+func TestWebhookScaling(t *testing.T) {
+	framework.NewTest(t).
+		// Limit to Kube environment as we're testing integration of webhook with K8s.
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			env := ctx.Environment().(*kube.Environment)
+			startGen := getVwcGeneration(vwcName, t, env)
+
+			// Scale up
+			scaleDeployment(istioNs, deployName, 2, t, env)
+			// Wait a bit to give the ValidatingWebhookConfiguration reconcile loop an opportunity to act
+			time.Sleep(sleepDelay)
+			gen := getVwcGeneration(vwcName, t, env)
+			if gen != startGen {
+				t.Fatalf("ValidatingWebhookConfiguration was updated unexpectedly on scale up")
+			}
+
+			// Scale down
+			scaleDeployment(istioNs, deployName, 1, t, env)
+			// Wait a bit to give the ValidatingWebhookConfiguration reconcile loop an opportunity to act
+			time.Sleep(sleepDelay)
+			gen = getVwcGeneration(vwcName, t, env)
+			if gen != startGen {
+				t.Fatalf("ValidatingWebhookConfiguration was updated unexpectedly on scale down")
+			}
+		})
+}
+
+// Verify that removing the galley deployment results in the webhook configuration being removed
+func TestWebhookRemoved(t *testing.T) {
```

Comment:
There is a clear need of ordering in this suite. This test is clearly meant to be executed as last. I'd suggest scoping everything, including the deployment of Istio to a single test method. Each subsequent test then can be a subtest within the context of the larger test. Here is an example:
https://github.com/ozevren/istio/blob/newgal-14-1-crd-support-test/tests/integration/galley/crd/crd_test.go

Reply:
I wondered about this. What I'd really like is isolation instead of ordering, so that we're resetting the istio install. At what level do we currently have that isolation? Suite level, perhaps? Is creating a new suite just for one test just too much overhead?

Reply:
I'm going ahead with (ordered) subtests for now. I'd still like a better understanding of whether or not tests are fully isolated at the suite level, and, if not, why that is.